README.md

```diff
@@ -1,17 +1,121 @@
+- [controller-v8.ts](api/facades/controller-v8.ts)
+- [controller-v9.ts](api/facades/controller-v9.ts)
+- [credential-manager-v1.ts](api/facades/credential-manager-v1.ts)
+- [firewall-rules-v1.ts](api/facades/firewall-rules-v1.ts)
+- [high-availability-v2.ts](api/facades/high-availability-v2.ts)
+- [image-manager-v2.ts](api/facades/image-manager-v2.ts)
+- [image-metadata-manager-v1.ts](api/facades/image-metadata-manager-v1.ts)
+- [key-manager-v1.ts](api/facades/key-manager-v1.ts)
+- [machine-manager-v6.ts](api/facades/machine-manager-v6.ts)
+- [metrics-debug-v2.ts](api/facades/metrics-debug-v2.ts)
+- [model-config-v2.ts](api/facades/model-config-v2.ts)
+- [model-generation-v4.ts](api/facades/model-generation-v4.ts)
+- [model-manager-v2.ts](api/facades/model-manager-v2.ts)
+- [model-manager-v3.ts](api/facades/model-manager-v3.ts)
+- [model-manager-v4.ts](api/facades/model-manager-v4.ts)
+- [model-manager-v5.ts](api/facades/model-manager-v5.ts)
+- [model-manager-v8.ts](api/facades/model-manager-v8.ts)
+- [payloads-v1.ts](api/facades/payloads-v1.ts)
+- [pinger-v1.ts](api/facades/pinger-v1.ts)
+- [resources-v1.ts](api/facades/resources-v1.ts)
+- [spaces-v6.ts](api/facades/spaces-v6.ts)
+- [sshclient-v2.ts](api/facades/sshclient-v2.ts)
+- [storage-v6.ts](api/facades/storage-v6.ts)
+- [subnets-v4.ts](api/facades/subnets-v4.ts)
+- [user-manager-v1.ts](api/facades/user-manager-v1.ts)
+- [user-manager-v2.ts](api/facades/user-manager-v2.ts)
+
+## Examples
+
+We have a number of examples showing how to perform a few common tasks. Those can be found in the examples folder.
+
+- [add-machine.js](examples/add-machine.js)
+- [deploy.js](examples/deploy.js)
+- [login-with-bakery.js](examples/login-with-bakery.js)
+- [ping.js](examples/ping.js)
+- [watch-all-models.js](examples/watch-all-models.js)
+- [watch.js](examples/watch.js)
+
+## Library Maintenance
+
+### Updating Library Facades
 
-## Updating the schema
+The Juju facade API files are generated from a supplied Juju schema. To update these facades this project provides a build script that can be run with `npm run build`. This will generate the facades using the schema stored at `$GOPATH/src/github.com/juju/juju/apiserver/facades/schema.json`. It will also pull the SHA and Juju version from this repository.
 
-To update the schema, run `make update-schema`. This will use the latest Juju
-develop as the source for generating a new schema and updating the schema.json
-file. A valid Go installation is required in order to run the command.
+### Releasing to NPM
 
-## Updating the API
+- Update the version number, respecting semver.
+- Run the tests with `npm test`.
+- Run the examples to ensure that everything works as expected.
+- Upgrade the [Juju Dashboard](https://github.com/canonical-web-and-design/jaas-dashboard) to this version using `npm link` and ensure that everything works as expected.
+- Tag the version in the git repository with the version number from above.
```

Comment:
Why not do a GitHub release which creates a tag and makes it easier for people to read release changes?

Reply:
I've actually always created the release from the tag, I didn't know it also worked the other way around :) Updated.